server/routes.go

```diff
@@ -955,6 +965,108 @@ func (s *Server) CreateBlobHandler(c *gin.Context) {
 	c.Status(http.StatusCreated)
 }
 
+func localBlobCopy (src, dest string) error {
+	_, err := os.Stat(src)
+	if err != nil {
+		return err
+	}
+
+	err = localCopy(src, dest)
+	if err == nil {
+		return nil
+	}
+
+	err = defaultCopy(src, dest)
+	if err == nil {
+		return nil
+	}
+
+	return fmt.Errorf("failed to copy blob")
+}
+
+func (s *Server) isLocal(c *gin.Context) bool {
+	if authz := c.GetHeader("Authorization"); authz != "" {
```

Comment:
As discussed in the prior PR, let's see if we can avoid full ssh key authentication here because on Linux the client and server have different keys. Instead let's see if we can:

1. Check the IP and request headers
2. If that fails (we should investigate a bit) we can try using a shared secret in a `/tmp` file